orca-keel/src/main/kotlin/com/netflix/spinnaker/orca/keel/task/DeleteIntentTask.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.keel.task
+
+import com.netflix.spinnaker.orca.ExecutionStatus
+import com.netflix.spinnaker.orca.KeelService
+import com.netflix.spinnaker.orca.RetryableTask
+import com.netflix.spinnaker.orca.TaskResult
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Component
+import java.util.HashMap
+import java.util.concurrent.TimeUnit
+
+@Component
+class DeleteIntentTask
+@Autowired constructor(
+  private val keelService: KeelService
+) : RetryableTask {
+  private val log = LoggerFactory.getLogger(javaClass)
+
+  override fun execute(stage: Stage) : TaskResult {
+    if (!stage.context.containsKey("intentId")) {
+      throw IllegalArgumentException("Missing required task parameter (intentId)")
+    }
+
+    val intentId = stage.context["intentId"].toString()
+
+    val status = if (stage.context.containsKey("status")) stage.context["status"].toString() else null
+
+    val response = keelService.deleteIntents(intentId, status)
+
+    val outputs = HashMap<String, Any>()
+    outputs.put("intent.id", intentId)
+
+    return TaskResult(
+      if (response.status == HttpStatus.NO_CONTENT.value()) ExecutionStatus.SUCCEEDED else ExecutionStatus.TERMINAL,
+      outputs
+    )
+  }
+
+  override fun getBackoffPeriod(): Long {
+    return TimeUnit.SECONDS.toMillis(15)
+  }
+
+  override fun getTimeout(): Long {
+    return TimeUnit.MINUTES.toMillis(1)
+  }
```

Comment:
Stylistic choice, but when returning a single value from an expression, I tend to just do something like:

`fun bleh() = TimeUnit.MINUTES.toMillis(1)`

Reply:
Thanks for all the kotlin tips and tricks and style choices. Please keep them coming!